common/app/navigation/Navigation.scala

```diff
@@ -0,0 +1,171 @@
+        case head :: tail => find(tail ++ head.children)
+      }
+    }
+
+    val maybeParent = find(children)
+
+    if(maybeParent.isDefined) {
+      maybeParent
+    } else {
+      // If the parent isn't found within the current edition, check other editions
+      find(getOtherPillarsFromEdition(edition))
+    }
+  }
+
+  def getPillar(pillars: Seq[NavLink], currentParent: NavLink, edition: Edition): NavLink = {
+    if(pillars.contains(currentParent)) {
+      currentParent
+    } else {
+      // TODO: should we have default pillar that is not UK (like a blank one)?
+      findParentByCurrentNavLink(currentParent, edition).getOrElse(ukNewsPillar)
+    }
+  }
+
+  def getSubnav(currentNavLink: Option[NavLink], currentParent: NavLink, currentPillar: NavLink): Subnav = {
+    val currentNavHasChildren = currentNavLink.map(navLink => navLink.children.nonEmpty).getOrElse(false)
+    val parentIsPillar =  currentNavLink.contains(currentPillar)
+
+    val showParent = if (parentIsPillar) false else true
+    val parent = if (!showParent) None else if(currentNavHasChildren) currentNavLink else Some(currentParent)
+
+    val childrenToShow = if (currentNavHasChildren) currentNavLink else Some(currentParent)
+    val children = childrenToShow.map( navLink => Some(navLink.children) ).getOrElse(None)
+
+    Subnav(parent, children, parent.isDefined, parent.isDefined && children.isDefined)
+  }
+}
+
+object NavRoot {
+  def apply(edition: Edition): NavRoot = {
+    edition match {
+      case editions.Uk => NavRoot(Seq(ukNewsPillar, ukSportPillar, ukOpinionPillar, ukArtsPillar, ukLifestylePillar))
+      case editions.Us => NavRoot(Seq(usNewsPillar, usSportPillar, usOpinionPillar, usArtsPillar, usLifestylePillar))
+      case editions.Au => NavRoot(Seq(auNewsPillar, auSportPillar, auOpinionPillar, auArtsPillar, auLifestylePillar))
+      case editions.International => NavRoot(Seq(auNewsPillar, auSportPillar, auOpinionPillar, auArtsPillar, auLifestylePillar))
+    }
+  }
+}
+
+case class SimpleMenu private (root: NavRoot) {
+  def pillars: Seq[NavLink] = root.children
+}
+
+case class NavMenu private (page: Page, root: NavRoot, edition: Edition) {
+
+  def currentUrl: String = NavMenu.getSectionOrPageUrl(page, edition)
+  def pillars: Seq[NavLink] = root.children
+
+  def currentNavLink: Option[NavLink] = root.findDescendantByUrl(currentUrl, edition).
+  def currentParent: NavLink = currentNavLink.flatMap( link => root.findParentByCurrentNavLink(link, edition) ).getOrElse(ukNewsPillar)
+  def currentPillar: NavLink = root.getPillar(pillars, currentParent, edition)
+
+  def subNavSections: Subnav = root.getSubnav(currentNavLink, currentParent, currentPillar)
+}
+
+object NavMenu {
+
+  def apply(page: Page, edition: Edition): NavMenu = NavMenu(page, NavRoot(edition), edition)
+
+  def apply(edition: Edition): SimpleMenu = SimpleMenu(NavRoot(edition))
+
+  def getSectionOrPageUrl(page: Page, edition: Edition): String = {
```

Comment:
This stuff is a bit ugly, but its just porting over the same logic as before and I would rather not rewrite this too right now